reservation/reservation_test.go

```diff
@@ -122,6 +122,28 @@ func TestReservationTTL(t *testing.T) {
 	assert.Nil(t, err)
 }
 
+func TestReservationWaitUntilLock(t *testing.T) {
+	manager, resourceID := setUp(t)
+
+	// Create a reservation and hold it for a little while
+	go func() {
+		reservation, err := manager.Lock(resourceID)
+		assert.Nil(t, err)
+		time.Sleep(5 * time.Second)
+		reservation.Release()
+	}()
+
+	// Assert that the reservation is currently held, and we can't get it
+	time.Sleep(time.Second)
```

Comment:
Can we use some kind of synchronization instead of sleeping? e.g. not trying to lock below until the goroutine has locked it, and then some message back to the goroutine to let it know to release the reservation